### PR TITLE
Convert Changelog to markdown, auto-commit future changelogs

### PR DESCRIPTION
**Pull request by qcasey**
## Proposed change

This PR accomplishes a few maintenance tasks:

1. Converts `changelog.rst` to `changelog.md`, enables markdown support in sphinx
2. Adapts CI to commit the generated release changelog to the docs on `main`
3. Manually adds the v1.7.1 changelog

This should save us the trouble of converting the generated changelog to restructuredText and manually commiting every release.

I would like to confirm if my [implementation in CI](https://github.com/qcasey/paperless-ngx/commit/7368411f97781092450b63c927657195b5eaf1a5) is correct? I've tested it over many ([many](https://github.com/qcasey/paperless-ngx/compare/main...convert-changelog-to-md)) iterations and it builds and releases correctly:

- [Successful CI release](https://github.com/qcasey/paperless-ngx/runs/6408946668?check_suite_focus=true)
- [Slightly tweaked release headings w/links](https://github.com/qcasey/paperless-ngx/releases/tag/v1.7.2)
- [Automated Changelog commit](https://github.com/qcasey/paperless-ngx/commit/997488201cd496d7b06c93102d3e327d020b8a4c)

Screenshot of the generated changelog in docs:

Screenshot from 2022-05-12 08-41-05

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain)

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by qcasey** [comment]
Also I do not know about the Pipfile.lock being updated so massively, that probably shouldn't be included, right?

File: Pipfile [hunk 1]
```diff
@@ -53,6 +53,7 @@ concurrent-log-handler = "*"
 zipp = {version = "*", markers = "python_version < '3.9'"}
 pyzbar = "*"
 pdf2image = "*"
+myst-parser = "*"
```
[the remark this answers is not in the document]
Comment (qcasey) [reply Pipfile]: I think that did the trick

**Comment by qcasey** [comment]
> This is great thanks! I dont love the rst format in general but this espeically is a big help for releases

Yeah I also prefer markdown. Although to its credit RST does [complicated tables](https://paperless-ngx.readthedocs.io/en/latest/scanners.html#physical-scanners) way better

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -381,3 +381,22 @@ jobs:
...
+        name: Checkout
+        uses: actions/checkout@v3
+      -
+        name: Append Changelog to docs
+        id: append-Changelog
+        working-directory: docs
+        run: |
+          git status
+          git pull origin main
+          git checkout main
```
Comment (stumpylog) [inline .github/workflows/ci.yml:394]: I think this could combine with the above checkout:
```
uses: actions/checkout@v3
with:
    ref: main
```

**Review by stumpylog — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (qcasey) [reply .github/workflows/ci.yml]: Yep good call, that appears to work well.

File: docs/changelog.md [hunk 3]
```diff
@@ -0,0 +1,1947 @@
...
+- Bump @<!---->ng-bootstrap/ng-bootstrap from 12.0.2 to 12.1.0 in /src-ui @dependabot (#834)
+- Bump pytest from 7.1.1 to 7.1.2 @dependabot (#806)
+- Bump github/codeql-action from 1 to 2 @dependabot (#792)
+- Bump imap-tools from 0.53.0 to 0.54.0 @dependabot (#758)
+- Bump ocrmypdf from 13.4.2 to 13.4.3 @dependabot (#757)
+- Bump importlib-resources from 5.6.0 to 5.7.1 @dependabot (#756)
+- Bump tox from 3.24.5 to 3.25.0 @dependabot (#692)
+- Bump cypress from 9.5.3 to 9.6.0 in /src-ui @dependabot (#800)
+- Bump angular \& tools to 13.3.4 or 13.3.3 @shamoon (#799)
+- Bump concurrently from 7.0.0 to 7.1.0 in /src-ui @dependabot (#797)
```
Comment (shamoon) [inline docs/changelog.md:68]:
```suggestion
- (chore) Runs pyupgrade to Python 3.8+ [\@stumpylog](https://github.com/stumpylog) ([\#890](https://github.com/paperless-ngx/paperless-ngx/pull/890))
- Dockerfile Organization \& Enhancements [\@stumpylog](https://github.com/stumpylog) ([\#888](https://github.com/paperless-ngx/paperless-ngx/pull/888))
- mobile friendlier manage pages [\@shamoon](https://github.com/shamoon) ([\#873](https://github.com/paperless-ngx/paperless-ngx/pull/873))
- Use semver for release process [\@stumpylog](https://github.com/stumpylog) ([\#851](https://github.com/paperless-ngx/paperless-ngx/pull/851))
- Enable Docker Hub push [\@stumpylog](https://github.com/stumpylog) ([\#828](https://github.com/paperless-ngx/paperless-ngx/pull/828))
- Feature barcode tiff support @gador ([\#766](https://github.com/paperless-ngx/paperless-ngx/pull/766))
- Updates GHA workflow to rebuild intermediate images on changes [\@stumpylog](https://github.com/stumpylog) ([\#820](https://github.com/paperless-ngx/paperless-ngx/pull/820))
- Adds simple Python to wait for Redis broker to be ready [\@stumpylog](https://github.com/stumpylog) ([\#788](https://github.com/paperless-ngx/paperless-ngx/pull/788))
- Update GHA workflow to build all Docker images [\@stumpylog](https://github.com/stumpylog) ([\#761](https://github.com/paperless-ngx/paperless-ngx/pull/761))

### Bug Fixes

- Feature / fix saved view \& sort field query params [\@shamoon](https://github.com/shamoon) ([\#881](https://github.com/paperless-ngx/paperless-ngx/pull/881))
- mobile friendlier manage pages [\@shamoon](https://github.com/shamoon) ([\#873](https://github.com/paperless-ngx/paperless-ngx/pull/873))
- Add timeout to healthcheck [\@shamoon](https://github.com/shamoon) ([\#880](https://github.com/paperless-ngx/paperless-ngx/pull/880))
- Always accept yyyy-mm-dd date inputs [\@shamoon](https://github.com/shamoon) ([\#864](https://github.com/paperless-ngx/paperless-ngx/pull/864))
- Fix local Docker image building [\@stumpylog](https://github.com/stumpylog) ([\#849](https://github.com/paperless-ngx/paperless-ngx/pull/849))
- Fix: show errors on invalid date input [\@shamoon](https://github.com/shamoon) ([\#862](https://github.com/paperless-ngx/paperless-ngx/pull/862))
- Fix: Older dates do not display on frontend [\@shamoon](https://github.com/shamoon) ([\#852](https://github.com/paperless-ngx/paperless-ngx/pull/852))
- Fixes IMAP UTF8 Authenication [\@stumpylog](https://github.com/stumpylog) ([\#725](https://github.com/paperless-ngx/paperless-ngx/pull/725))
- Fix password field remains visible [\@shamoon](https://github.com/shamoon) ([\#840](https://github.com/paperless-ngx/paperless-ngx/pull/840))
- Fixes Pillow build for armv7 [\@stumpylog](https://github.com/stumpylog) ([\#815](https://github.com/paperless-ngx/paperless-ngx/pull/815))
- Update frontend localization source file [\@shamoon](https://github.com/shamoon) ([\#814](https://github.com/paperless-ngx/paperless-ngx/pull/814))
- Fix install script extra OCR languages format [\@stumpylog](https://github.com/stumpylog) ([\#777](https://github.com/paperless-ngx/paperless-ngx/pull/777))

### Documentation

- Use semver for release process [\@stumpylog](https://github.com/stumpylog) ([\#851](https://github.com/paperless-ngx/paperless-ngx/pull/851))
- Deployment: Consolidate tika compose files @qcasey ([\#866](https://github.com/paperless-ngx/paperless-ngx/pull/866))
- Fix local Docker image building [\@stumpylog](https://github.com/stumpylog) ([\#849](https://github.com/paperless-ngx/paperless-ngx/pull/849))

### Maintenance

- Dockerfile Organization \& Enhancements [\@stumpylog](https://github.com/stumpylog) ([\#888](https://github.com/paperless-ngx/paperless-ngx/pull/888))
- Add timeout to healthcheck [\@shamoon](https://github.com/shamoon) ([\#880](https://github.com/paperless-ngx/paperless-ngx/pull/880))
- Use semver for release process [\@stumpylog](https://github.com/stumpylog) ([\#851](https://github.com/paperless-ngx/paperless-ngx/pull/851))
- Deployment: Consolidate tika compose files @qcasey ([\#866](https://github.com/paperless-ngx/paperless-ngx/pull/866))
- Fixes Pillow build for armv7 [\@stumpylog](https://github.com/stumpylog) ([\#815](https://github.com/paperless-ngx/paperless-ngx/pull/815))
- Update frontend localization source file [\@shamoon](https://github.com/shamoon) ([\#814](https://github.com/paperless-ngx/paperless-ngx/pull/814))
- Fix install script extra OCR languages format [\@stumpylog](https://github.com/stumpylog) ([\#777](https://github.com/paperless-ngx/paperless-ngx/pull/777))
- Adds simple Python to wait for Redis broker to be ready [\@stumpylog](https://github.com/stumpylog) ([\#788](https://github.com/paperless-ngx/paperless-ngx/pull/788))

### Dependencies

<details>
<summary>15 changes</summary>

- Bump tj-actions/changed-files from 18.7 to 19 @dependabot ([\#830](https://github.com/paperless-ngx/paperless-ngx/pull/830))
- Bump asgiref from 3.5.0 to 3.5.1 @dependabot ([\#867](https://github.com/paperless-ngx/paperless-ngx/pull/867))
- Bump jest from 27.5.1 to 28.0.3 in /src-ui @dependabot ([\#860](https://github.com/paperless-ngx/paperless-ngx/pull/860))
- Bump @<!---->ng-bootstrap/ng-bootstrap from 12.1.0 to 12.1.1 in /src-ui @dependabot ([\#861](https://github.com/paperless-ngx/paperless-ngx/pull/861))
- Bump @<!---->types/node from 17.0.27 to 17.0.29 in /src-ui @dependabot ([\#833](https://github.com/paperless-ngx/paperless-ngx/pull/833))
- Bump @<!---->ng-bootstrap/ng-bootstrap from 12.0.2 to 12.1.0 in /src-ui @dependabot ([\#834](https://github.com/paperless-ngx/paperless-ngx/pull/834))
- Bump pytest from 7.1.1 to 7.1.2 @dependabot ([\#806](https://github.com/paperless-ngx/paperless-ngx/pull/806))
- Bump github/codeql-action from 1 to 2 @dependabot ([\#792](https://github.com/paperless-ngx/paperless-ngx/pull/792))
- Bump imap-tools from 0.53.0 to 0.54.0 @dependabot ([\#758](https://github.com/paperless-ngx/paperless-ngx/pull/758))
- Bump ocrmypdf from 13.4.2 to 13.4.3 @dependabot ([\#757](https://github.com/paperless-ngx/paperless-ngx/pull/757))
- Bump importlib-resources from 5.6.0 to 5.7.1 @dependabot ([\#756](https://github.com/paperless-ngx/paperless-ngx/pull/756))
- Bump tox from 3.24.5 to 3.25.0 @dependabot ([\#692](https://github.com/paperless-ngx/paperless-ngx/pull/692))
- Bump cypress from 9.5.3 to 9.6.0 in /src-ui @dependabot ([\#800](https://github.com/paperless-ngx/paperless-ngx/pull/800))
- Bump angular \& tools to 13.3.4 or 13.3.3 [\@shamoon](https://github.com/shamoon) ([\#799](https://github.com/paperless-ngx/paperless-ngx/pull/799))
- Bump concurrently from 7.0.0 to 7.1.0 in /src-ui @dependabot ([\#797](https://github.com/paperless-ngx/paperless-ngx/pull/797))
```

**Comment by shamoon** [comment]
I didnt notice any issues with building the docs, etc. I just added a comment for putting links to usernames / PRs in the 1.7.1 changelog section

**Review by shamoon — APPROVED** [review]

**Review by stumpylog — APPROVED** [review]

**Comment by tooomm** [comment]
It looks like readthedocs is not picking it up yet? https://paperless-ngx.readthedocs.io/en/latest/changelog.html

**Comment by tooomm** [comment]
Thanks for the quick fix! :)

However, I noticed another minor issue: The linking of the PR creator is not always working:
Untitled
On the GH release page (https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.7.1) they show.

**Comment by shamoon** [comment]
> However, I noticed another minor issue: The linking of the PR creator is not always working

From now on those will be auto-generated, those ones were manual, I can fix that though obviously not really a big deal =)

**Comment by tooomm** [comment]
Oh ok, I thought the auto generation might not properly work.

I can take care of the linking tomorrow, I want to add the same subsections to 1.7 and 1.6 as well to make it more uniform anyway. 👍

**Comment by shamoon** [comment]
Ah I already did that. I dont think we need collapsable sections for minor versions tbh, one want to bury things too much. IMHO, anyway.

**Comment by tooomm** [comment]
> I dont think we need collapsable sections for minor versions tbh, one want to bury things too much. IMHO, anyway.

Can you clarify this, please? @shamoon 
Right now the "minor" releases 1.6.0 and 1.7.0 do not have collapsable subgroups, but the "patch" release 1.7.1 does.

So you prefer to not have them there either?

**Comment by shamoon** [comment]
Oh, I guess those came along with the change to using auto-generated changelogs / markdown, i.e. the changes in this PR. I hadn't noticed. I dunno, dont feel strongly about it but I think theyre fine. What I meant was I didnt want to bury the -ngx versions another level down. Again, dont feel very strongly about this stuff, just my opinion

**Comment by tooomm** [comment]
>What I meant was I didnt want to bury the -ngx versions another level down.

Oh, that was totally not my intention. I like that the development of the former projects is available, but they don't need to be as present and can be combined like that. I would definitely not do that for the main -ngx project and "bury them another level down". Agreed.

I was striving for some more uniformity within ngx changelogs for easier readability and coherent navigation with this automation in place. Will open a PR.

**Pull request merged**